models/articles.model.js

```
@@ -37,26 +37,69 @@ exports.fetchAllArticles = async (sort_by = "created_at", order = "ASC") => {
     }
 };
 
-
 exports.fetchArticleById = async (article_id) => {
     try {
-        let baseQuery = `
+        const articleQuery = `
      SELECT a.article_id, a.title, a.body, a.created_at, a.votes, a.article_img_url,
        u.username AS author, t.slug AS topic
 FROM articles a
 LEFT JOIN users u ON a.author = u.username
 LEFT JOIN topics t ON a.topic = t.slug
-WHERE a.article_id = $1;`
-
+WHERE a.article_id = $1; 
+`
 
         const queryParams = [article_id];
-        const result = await db.query(baseQuery, queryParams);
 
-        if (!result.rows.length) {
-            console.log("No article found with the given ID");
+        const articlesResult = await db.query(articleQuery, queryParams);
+
+        if (!articlesResult.rows.length) {
             throw new Error("Article not found");
         }
-        return result.rows[0];
+        return articlesResult.rows[0];
 
     } catch (err) { }
-}
+};
+
+exports.fetchArticleCommentsById = async (article_id) => {
+    try {
+
+        const commentsQuery = `SELECT c.body, c.comment_id, c.votes, c.created_at,
+         c.article_id, 
+               c.author
+FROM comments c
+WHERE c.article_id = $1
+ORDER BY created_at DESC
+`
+
+        const queryParams = [article_id];
+
+        const commentsResult = await db.query(commentsQuery, queryParams);
+
+        if (!commentsResult.rows.length) {
+            return []
+        }
+
+        return commentsResult.rows;
+
+    } catch (err) {
+    }
+};
+
+exports.insertComments = async (article_id, body, username) => {
+    try {
+        const insertCommentsQuery = `
+ INSERT INTO comments 
+ (article_id, body, author) 
+         VALUES ($1, $2, $3)
+         RETURNING comment_id, article_id, body, author, created_at;`
+
+        const commentValues = [article_id, body, username];
+
+        const insertResult = await db.query(insertCommentsQuery, commentValues);
+
+        return insertResult.rows[0];
+
+    } catch (err) {
+        throw err
```

I'm not sure why you'd need to throw an error here - an error will come back from PSQL and it will be caught in the controller. This whole section doesn't need to happen with a try/catch :)